### PR TITLE
feat(UI): add quotes when tests pass

### LINKED ISSUE #50634: Motivational quotes might be confusing
Carrying on from #46429:

Some of the `compliments` in our [quotes object](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/i18n/locales/english/motivation.json), such as `They're in another castle`, may not make sense to campers.

Some of these are pop-culture references, like `"Legen - wait for it - dary!",` and `"Moon Prism Power!"`. As someone who is familiar with those two particular phrases, seeing them pop up in my learning journey was a nice treat. However, campers who are not familiar with the references might not understand the meaning here.

Do we want to go through and clean up the ones that don't directly compliment the camper?

Original Conversation: https://github.com/freeCodeCamp/freeCodeCamp/pull/46429#issuecomment-1581267853

### PR DESCRIPTION
**Pull request by JordanMooree**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref #46178 
closes: #50634

> Accidentally ran `git push -f` on previous branch for PR #46429 - Added the e2e test as suggested [here](https://github.com/freeCodeCamp/freeCodeCamp/pull/46429#issuecomment-1598267031).

- Fixed typo in file name for `quote.tsx`
- Ran `pnpm run cypress run --spec=cypress/e2e/default/learn/challenges/lower-jaw.ts`:

image

### CODE REVIEW

**Comment by JordanMooree** [comment]
Let me know if there are any more test cases you think should be handled and I could add them.

File: cypress/e2e/default/learn/challenges/lower-jaw.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+describe('lower jaw', () => {
+  before(() => {
+    cy.visit(outputLocation.catApp);
+  });
+
+  it(
+    'Should show the quote when the code passes',
+    { browser: 'electron' },
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/challenges/lower-jaw.ts:14]: Why only run this on electron?

File: cypress/e2e/default/learn/challenges/lower-jaw.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      cy.visit(
+        '/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-2'
+      );
+      cy.get(`${'.react-monaco-editor-container'} textarea`, { timeout: 16000 })
+        .click()
+        .focused()
+        .type('{downArrow}')
+        .clear()
+        .type('<h2>Cat Photos</h2>');
+      cy.contains('Check Your Code (Ctrl + Enter)').click({ force: true });
```
Comment (naomi-lgbt) [inline cypress/e2e/default/learn/challenges/lower-jaw.ts:25]: Do we need to force the click here?

**Review by naomi-lgbt — DISMISSED** [review]

File: cypress/e2e/default/learn/challenges/lower-jaw.ts (hunk 2 above)
Comment (JordanMooree) [reply cypress/e2e/default/learn/challenges/lower-jaw.ts]: Good catch, fixed!

File: cypress/e2e/default/learn/challenges/lower-jaw.ts (hunk 1 above)
Comment (JordanMooree) [reply cypress/e2e/default/learn/challenges/lower-jaw.ts]: I was following a similar pattern I found in `projects` cypress test. I've updated to include more browsers in the fix (`'electron', 'chrome', 'chromium', 'firefox', 'edge'`).

Comment (ahmaxed) [reply cypress/e2e/default/learn/challenges/lower-jaw.ts]: I just removed them all-together.

**Comment by ahmaxed** [comment]
I went ahead and gave it an id so it is not confused with the hint description.

**Review by ahmaxed — APPROVED** [review]

**Review by naomi-lgbt — COMMENTED** [review]
This is looking good - before we get this in, I want to confirm with @bbsmooth on the accessibility.

image


When using a screen reader and arrowing down, I hear "Congratulations, your code passes. Submit your code to continue." Arrow down again and I hear "Tubular!".

Do we want to provide any context to the quote?

**Comment by bbsmooth** [comment]
> Do we want to provide any context to the quote?

This is one of those questions where you really need to do some user testing to get a good answer. Otherwise, you try your best and rely on feedback. Personally, I think the way we have it now is OK, but I'm not a native screen reader user, so I may be wrong.

We use the check mark icon as a visual signal that the step passed. But we also have text that says "Congratulations, your code passes", which is a text alternative to the icon, so the icon basically becomes irrelevant. In other words, the check mark icon is not needed in order to understand that the step was passed.

We do have a quote icon next to the quote. I guess the question is, is that quote icon necessary to understand the context of the text quote? If not, then we probably don't need a text alternative for it. If so, then we probably do. Personally, I always thought it was just there to match up with the check mark icon and considered it decorative. I'd be interested to hear what message other people think the icon is conveying. Or maybe even with the quote icon it is still not clear to sighted users what the quote represents? This could perhaps be more than just a screen reader only issue.

Long story short, I think we are OK with what we have now, but I'm leaving room open for different opinions or future feedback from screen reader users. Such is the way with accessibility. Sometimes there is no one right answer.

**Review by Sboonny — APPROVED** [review]
